libdnf/rpm/repo_sack.cpp

```diff
@@ -30,4 +35,93 @@ RepoWeakPtr RepoSack::new_repo(const std::string & id, Repo::Type type) {
     return add_item_with_return(std::move(repo));
 }
 
+static void load_config_from_parser(
+    Config & conf, const ConfigParser & parser, const std::string & section, Base & base) {
+    auto & logger = base.get_logger();
+    const auto & cfg_parser_data = parser.get_data();
+    auto cfg_parser_data_iter = cfg_parser_data.find(section);
+    if (cfg_parser_data_iter != cfg_parser_data.end()) {
+        auto opt_binds = conf.opt_binds();
+        const auto & cfg_parser_sect = cfg_parser_data_iter->second;
+        for (const auto & opt : cfg_parser_sect) {
+            auto opt_binds_iter = opt_binds.find(opt.first);
+            if (opt_binds_iter != opt_binds.end()) {
+                try {
+                    auto value = opt.second;
+                    parser.substitute(value, base.get_variables());
+                    opt_binds_iter->second.new_string(Option::Priority::MAINCONFIG, value);
+                } catch (const Option::Exception & ex) {
+                    auto msg = fmt::format(
+                        R"**(Config error in section "{}" key "{}": {}: {})**",
+                        section,
+                        opt.first,
+                        ex.get_description(),
+                        ex.what());
+                    logger.warning(msg);
+                }
+            }
+        }
+    }
+}
+
+void RepoSack::new_repos_from_file(const std::string & path) {
+    auto & logger = base->get_logger();
+    ConfigParser parser;
+    parser.read(path);
+    const auto & cfg_parser_data = parser.get_data();
+    for (const auto & cfg_parser_data_iter : cfg_parser_data) {
+        const auto repo_id = cfg_parser_data_iter.first;
+        if (repo_id != "main") {
+            auto bad_char_idx = Repo::verify_id(repo_id);
+            if (bad_char_idx >= 0) {
+                logger.warning(
+                    fmt::format("Bad id for repo: {}, byte = {} {}", repo_id, repo_id[bad_char_idx], bad_char_idx));
+                continue;
+            }
+
+            auto repo = new_repo(repo_id);
+            auto repo_cfg = repo->get_config();
+            logger.debug(fmt::format(R"**(Start of loading configuration of repository "{}" from file "{}")**", repo_id, path));
+            load_config_from_parser(*repo_cfg, parser, repo_id, *base);
+            logger.debug(fmt::format(R"**(Loading configuration of repository "{}" from file "{}" done)**", repo_id, path));
```

Comment:
We have to discuss how we want to handle logging (philosophy). Like what we want to log and what to show to user.

Reply:
OK. I think that is good idea to log debug message in some points. The debug messages in the log have nothing to do with standard user messages on the console.
Let's talk about it.